bcda/servicemux/servicemux_test.go

```diff
@@ -18,7 +18,7 @@ type ServiceMuxTestSuite struct {
 }
 
 func (s *ServiceMuxTestSuite) TestNew() {
-	sm := New("127.0.0.1:0")
+	sm := New(os.Getenv("SM_TEST_ADDR"))
```

Comment:
I get that the quality check in Sonar is no hard-coded IPs, but later on in the same test there's an assertion that sm == "127.0.0.1". It seems the test[s] expect this to be the loopback address, in which case why not just create a const LoopbackAddress (or something like that) and use it? This way, we can minimize the change and not have to change docker configs here and in ops.

Reply:
yeah, i thought about that.  using a const still counts as a hard-coded IP address though.  I could use a config file for just this test, even though that seems a little silly.  i was trying to find somewhere that the loopback address was defined in http or a similar package but no dice so far.

Reply:
Often there is a way to express an exception to a rule by putting a comment in the code. Do we know what inspection tool Sonar is using, and can we find it's exception notation and use it here?

Reply:
There doesn't seem to be a way to ignore individual lines with comments in sonarqube.  only full files and directories

Reply:
i ended up going with a json config file to hold the ip address